### PR TITLE
Linear interp with NaNs in nd indexer

### PR DESCRIPTION
**Pull request by jenssss**
- [x] Test added

When doing linear interpolatiion with an nd indexer that contains NaN's,  `xarray` previously threw a `KeyError` from the `missing._localize` function. This PR fixes this by swapping `np.min` and `np.max` with `np.nanmin` and `np.nanmax` in that function, ignoring any NaN values.

### CODE REVIEW

**Comment by fujiisoup** [comment]
Thanks, @jenssss for sending a PR.
This looks good to me.
Could you add a line for this contribution to our whatsnew?

File: xarray/tests/test_interp.py [hunk 1]
```diff
@@ -274,6 +274,25 @@ def test_interpolate_nd_nd():
         da.interp(a=ia)
 
 
+def test_interpolate_nd_with_nan():
```
Comment (fujiisoup) [inline xarray/tests/test_interp.py:277]: You need `@requires_scipy` for an environment without scipy installed.

File: xarray/core/missing.py [hunk 2]
```diff
@@ -560,8 +560,8 @@ def _localize(var, indexes_coords):
     indexes = {}
     for dim, [x, new_x] in indexes_coords.items():
         index = x.to_index()
-        imin = index.get_loc(np.min(new_x.values), method="nearest")
-        imax = index.get_loc(np.max(new_x.values), method="nearest")
+        imin = index.get_loc(np.nanmin(new_x.values), method="nearest")
+        imax = index.get_loc(np.nanmax(new_x.values), method="nearest")
```
Comment (fujiisoup) [inline xarray/core/missing.py:564]: It looks that np.nanmin (and nanmax) supports np.datetime-dtype only with numpy>=1.18.
We can copy [np.nanmin](https://github.com/numpy/numpy/blob/v1.19.0/numpy/lib/nanfunctions.py#L228-L336) to our core/npcompat.py and call this function here.

Comment (mathause) [reply xarray/core/missing.py:564]: I think they added some low level loops to fix this (numpy/numpy#14841) so guess we cannot copy nanmin over and have to use `if LooseVersion(np. __version__) < LooseVersion("1.18")` (or similar).

**Comment by mathause** [comment]
Looks good - if you want to go a step further you can probably do something along the lines of (untested):

```python
if new_x.dtype in "mM" and LooseVersion(np.__version__) < LooseVersion("1.18") and new_x.isnull().any():
    raise ValueError("numpy 1.18 or newer required to use interp with datetime/ timedelta array containing missing values")
else:
    imin = ... np.nanmin(...)
    imax = ...
```

**Comment by mathause** [comment]
I misremembered the behavior of the old `np.min` with datetimes - see this comment: https://github.com/pydata/xarray/pull/3924/files#discussion_r407101544

Sorry for sending you down the wrong path. I can have another look tomorrow.

**Comment by jenssss** [comment]
Okay, I removed the `ValueError` from my previous commit, so the content of `_localize` is now basically the same as in https://github.com/pydata/xarray/pull/3924/files#discussion_r407101544

I thought it better to use `np.issubdtype(new_x.dtype, np.datetime64)` instead of  `new_x.dtype.kind in "mM"` to check for datetime type. It's a bit longer, but I think it makes it more readable since you don't have to look up what `dtype.kind` is.

**Comment by mathause** [comment]
Thanks! I have two more suggestions:

Can you add an additional line with 
```python
(["2000-01-01T12:00", "2000-01-02T12:00", "NaT"], [0.5, 1.5]), 
```
here:

https://github.com/pydata/xarray/blob/8313c3eb7e92c15a82a3ae8ba7dff7d49a37c309/xarray/tests/test_interp.py#L575

to have a test with a missing datetime.

Do you also want to add a note to the docstring, e.g. adding `Missing values are skipped`. Here:

https://github.com/pydata/xarray/blob/a36d0a1d4657c848dcdd76d0ecb9c783ad464057/xarray/core/dataarray.py#L1425
here:
https://github.com/pydata/xarray/blob/a36d0a1d4657c848dcdd76d0ecb9c783ad464057/xarray/core/dataarray.py#L1495

Does it also work for DataSets? Then maybe add to the docstring here:

https://github.com/pydata/xarray/blob/a36d0a1d4657c848dcdd76d0ecb9c783ad464057/xarray/core/dataset.py#L2603

and here:

https://github.com/pydata/xarray/blob/a36d0a1d4657c848dcdd76d0ecb9c783ad464057/xarray/core/dataset.py#L2731

**Comment by mathause** [comment]
LGTM. I'll merge in a day or two unless someone else has a comment.

**Comment by mathause** [comment]
thanks @jenssss I see this is your first PR - welcome to xarray!

**Pull request merged**